### ISSUE TITLE
Preview is stopped working after upgrade from version 24 to 25.

### ISSUE THREAD

**Issue by kevinkuan1969** [nextcloud/previewgenerator#332]

**Comment by kevinkuan1969** [comment]
The Preview was working perfectly well before upgrade to version 25. After upgrade for 1 week, the Preview is not working entirely. Found all the answers and try it out but still cannot solve it. 

We have tried the following methods:
 - re-run the occ preview:generate-all -vvv
 - re-run the occ files:scan-app-data
 - Delete records from database
    mysql> delete from oc_filecache where path like "appdata_ocm2viqqd6xv/preview/%"; 
- Delete the appdata folder root@cloud:/var/www/nextcloud/data/appdata_ocm2viqqd6xv# rm -rf preview
- re-run preview generate command sudo -u www-data php occ preview:generate-all -vvv
- ensure all the php is installed such as php-cli php-mysql php-gd php-xml php-mbsting php-zip and php-curl
Still no clue and hope someone can help. Thanks

**Comment by kevinkuan1969** [comment]
Upgrade Nextcloud 24 to Nextcloud 25.0.2

**Comment by st3iny** [comment]
That is unfortunate but I can't reproduce it. I need more information to debug this. 

- How about the list list enabled preview providers from config.php? 
- Which OS are you using? 
- Is this a dockerized setup? 
- What do you mean by "Preview is not working entirely"? Are there errors when running occ preview:* commands or are no previews shown in the web ui?

**Comment by t-markmann** [comment]
I have a similar issue.
NC 25.0.5.1
Current Apache with PHP 8.0

```
sudo -u www-data /var/www/nextcloud/occ preview:generate-all -vvv

In Application.php line 676:

  [Symfony\Component\Console\Exception\CommandNotFoundException]
  Command "preview:generate-all" is not defined.

  Did you mean one of these?
      notification:generate
      preview:repair
      preview:reset-rendered-texts


Exception trace:
  at /var/www/nextcloud/3rdparty/symfony/console/Application.php:676
 Symfony\Component\Console\Application->find() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:237
 Symfony\Component\Console\Application->doRun() at /var/www/nextcloud/3rdparty/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /var/www/nextcloud/lib/private/Console/Application.php:214
 OC\Console\Application->run() at /var/www/nextcloud/console.php:100
 require_once() at /var/www/nextcloud/occ:11
```

Tried with and without pre-providers in config.php
```
  'enabledPreviewProviders' =>
  array (
    0 => 'OC\\Preview\\PNG',
    1 => 'OC\\Preview\\JPEG',
    2 => 'OC\\Preview\\GIF',
    3 => 'OC\\Preview\\HEIC',
    4 => 'OC\\Preview\\BMP',
    5 => 'OC\\Preview\\XBitmap',
    6 => 'OC\\Preview\\MP3',
    7 => 'OC\\Preview\\TXT',
    8 => 'OC\\Preview\\MarkDown',
    9 => 'OC\\Preview\\OpenDocument',
    10 => 'OC\\Preview\\Krita',
    11 => 'OC\\Preview\\TIFF',
  ),
```

Related to https://github.com/nextcloud/previewgenerator/issues/246 ?

Last time this worked was with NC 25.0.3. 
Yesterday I upgraded to 25.0.5.

Preview-generator App is installed and I even reinstalled. Still does not work.

**Comment by James-Oakley** [comment]
The issue is that you're using PHP 8.0. When I upgraded from NC24.x to NC25.x, the Apps settings screen in my installation shows that the Preview Generator app has been disabled. 

> This app cannot be installed because the following dependencies are not fulfilled:
> 
> -     PHP with a version lower than 8.0 is required. 

I'm not sure why this wasn't a problem before NC25.x - I have been using PHP 8.1 quite happily for a while with NextCloud ever since the main NextCloud system supported it. PHP 7.4 has been EOL since November 2022, so any web application today needs to support PHP 8.0 as a minimum.

So I'm unsure of the solution, but I think the upgrade script uninstalling the app is the cause of the problem

**Comment by James-Oakley** [comment]
So in my case, I checked the head of the code, saw it fully supported more recent PHP, and realised that I was several versions behind for this app. Checking out to the latest tag fixed the problems

**Comment by st3iny** [comment]
That is odd. I pushed a fixed release that adds the support for PHP 7.4 again as it's supported by Nextcloud 25. It should support PHP v7.4 up to v8.2.

Try to reinstall/update the app and make sure that it tries to install previewgenerator v5.3.0 which is the most recent release.

**Comment by James-Oakley** [comment]
> That is odd. I pushed a fixed release that adds the support for PHP 7.4 again as it's supported by Nextcloud 25. It should support PHP v7.4 up to v8.2.
> 
> Try to reinstall/update the app and make sure that it tries to install previewgenerator v5.3.0 which is the most recent release.

I'm now all happy. My problem, I think, was that I installed it into /apps originally using git clone instead of the app manager within the installation. As a result, it hadn't updated for ages, because I was being left to do that manually. Running git checkout on the latest release tag fixed it all

**Comment by st3iny** [comment]
Closing as this seems to be resolved now.
